hammerblade/torch/kernel/kernel_embedding_backward.cpp

```diff
@@ -0,0 +1,54 @@
+//====================================================================
+// Embedding backward kernel
+// 04/22/2020 Lin Cheng (lc873@cornell.edu)
+//====================================================================
+
+#include <kernel_common.hpp>
+
+extern "C" {
+
+  __attribute__ ((noinline))  int tensorlib_embedding_backward(
+          hb_tensor_t* grad_weight_p,
+          hb_tensor_t* grad_p,
+          hb_tensor_t* index_p,
+          int32_t* padding_idx_p,
+          int32_t* num_weights_p,
+          int32_t* numel_p) {
+
+    HBTensor<float> grad_weight(grad_weight_p);
+    HBTensor<float> grad(grad_p);
+    HBTensor<int32_t> index(index_p);
+    int32_t padding_idx = *padding_idx_p;
+    int32_t num_weights = *num_weights_p;
+    int32_t numel = *numel_p;
+    float* grad_weight_data = (float*)grad_weight.data_ptr();
+    float* grad_data = (float*)grad.data_ptr();
+    int32_t* index_data = (int32_t*)index.data_ptr();
```

Comment:
This might be nitpick... isn't this one of kernels where we should use raw tensors? Apparently HBTensor only methods are not called.

Reply:
The hope is (in the future) there will be auto-generated wrappers to hide hb_tensor_t interface. Kernel writers will only deal with HBTensors.